lib/yard/docstring_parser.rb

```diff
@@ -224,7 +224,8 @@ def create_tag(tag_name, tag_buf = '')
 
     # Creates a {Tags::RefTag}
     def create_ref_tag(tag_name, name, object_name)
-      @tags << Tags::RefTagList.new(tag_name, P(object, object_name), name)
+      namespace = object && object.docstring ? object.docstring.namespace : object
+      @tags << Tags::RefTagList.new(tag_name, P(namespace, object_name), name)
```

Comment:
(See above) I could be wrong but it seems like `namespace` could be replaced entirely by `object || @unresolved_reference` and thus be the only necessary line of implementation.